### PR TITLE
SONIC LLDP interface model definition

### PR DESCRIPTION
**Pull request by divya-balasubramania**
LLDP interface model definition for SONIC

### CODE REVIEW

File: models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+              network_policy:
+                description:
+                     - This command can be used to select whether to advertise network-policy
+                       LLDP-MED TLVs or not.By default network-policy LLDP-MED TLVs are advertised.
+                type: str
+              power_management:
+                description:
+                     - This command can be used to select whether to advertise power-management
+                       LLDP-MED TLVs or not.By default power-management LLDP-MED TLVs are advertised.
+                type: str
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml:64]: Please change the type to `bool` for enabling or disabling the options.

File: models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          tlv_select:
+             description:
+               - This command can be used to select whether to advertise the LLDP 802.3at or bt
+                 power management TLVs or not. By default this TLV is advertised.
+               - This command is supported only on physical interfaces and not on logical interfaces.
+             type: str
+             choices:
+                - 'power-management'
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml:72]: Please change "tlv_select" to `list` and "power_management" as one of its sub-options of type `bool`, similar to structure of "med_tlv_select"

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]

File: models/enterprise_sonic/lldp_interfaces/deleted_example_02.txt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# !
+# sonic#
+
+  - name: Delete default LLDP Interface configurations
+    dellemc.enterprise_sonic.sonic_lldp_interfaces:
+      config:
+        tlv_select:
+          power-management: True
+        med_tlv_select:
+          network_policy: True
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/deleted_example_02.txt:27]: Please include the name of the interface to be configured.

File: models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            choices:
+               - 'receive'
+               - 'transmit'
+          med_tlv_select:
+            description:
+              - This command can be used to select whether to advertise the LLDP-MED TLVs or not.
+                By default the LLDP-MED TLVs are advertised.
+              - This command is supported only on physical interfaces and not on logical interfaces.
+            type: list
+            elements: dict
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml:53]: Please change type to 'dict'.

File: models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                     - This command can be used to select whether to advertise power-management
+                       LLDP-MED TLVs or not.By default power-management LLDP-MED TLVs are advertised.
+                type: bool
+          tlv_select:
+            description:
+               - This command can be used to select whether to advertise the LLDP 802.3at or bt
+                 power management TLVs or not. By default this TLV is advertised.
+               - This command is supported only on physical interfaces and not on logical interfaces.
+            type: list
+            elements: dict
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml:71]: Please change type to 'dict'.

File: models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                     - This command can be used to select whether to advertise power-management
+                       LLDP TLVs or not.By default power-management LLDP TLVs are advertised.
+                type: bool
+          tlv_set:
+             description:
+               -   This command can be used to configure an IPv4 or IPv6 management address
+                that will be used to advertise by LLDP on an interface
+               - This command is supported only on physical interfaces and not on logical interfaces.
+             type: list
+             elements: dict
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml:84]: Please change type to 'dict'.

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]

File: models/enterprise_sonic/lldp_interfaces/deleted_example_01.txt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# interface Ethernet1
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  lldp transmit
+#  lldp tlv-set management-address ipv4 10.1.1.2
+# sonic#
+#
+  - name: Delete LLDP interface configurations
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/deleted_example_01.txt:16]:
```suggestion
# sonic#

  - name: Delete LLDP interface configurations
```

File: models/enterprise_sonic/lldp_interfaces/deleted_example_02.txt
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/deleted_example_02.txt:1]: Please update the after and before state to list the same set of interfaces.

File: models/enterprise_sonic/lldp_interfaces/deleted_example_03.txt [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+#  unreliable-los auto
+#  no shutdown
+#  lldp receive
+#  lldp tlv-set management-address ipv4 10.1.1.2
+#  no lldp med-tlv-select network-policy
+#  no lldp tlv-select power-management
+# sonic#
+
+
+  - name: Delete all LLDP Interface configuration
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/deleted_example_03.txt:19]:
```suggestion
# sonic#

  - name: Delete all LLDP Interface configurations
```

File: models/enterprise_sonic/lldp_interfaces/deleted_example_03.txt
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/deleted_example_03.txt:1]: Please list multiple interfaces in before and after state, to show the deletion of non-default LLDP configurations in all the interfaces with an empty config.

File: models/enterprise_sonic/lldp_interfaces/merged_example_01.txt
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/merged_example_01.txt:1]: Please update the after and before state to list the same set of interfaces.

File: models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+      config:
+        description: The set of link layer discovery protocol interface attribute configurations
+        type: list
+        elements: dict
+        suboptions:
+          name:
+            description:
+              - Interface name in which LLDP needs to be configured on.
+            type: str
+            required: True
```
Comment (ArunSaravananBalachandran) [inline models/enterprise_sonic/lldp_interfaces/sonic_lldp_interfaces.yml:32]:
```suggestion
            required: true
```

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]
Please move the delete all example to 'deleted_example_04.txt' and include a new example in 'deleted_example_03.txt' to showcase deletion of all non-default LLDP configurations in an interface by specifying only the name of that interface.

Also, please ensure that all the bool values in the examples are in the format `true`/`false`.

File: models/enterprise_sonic/lldp_interfaces/deleted_example_01.txt (hunk 7 above)
Comment (divya-balasubramania) [reply models/enterprise_sonic/lldp_interfaces/deleted_example_01.txt]: done

File: models/enterprise_sonic/lldp_interfaces/deleted_example_02.txt
Comment (divya-balasubramania) [reply models/enterprise_sonic/lldp_interfaces/deleted_example_02.txt:1]: done

**Review by ArunSaravananBalachandran — APPROVED** [review]

**Pull request merged**